mpv-drmprime: enable Lua support for support of built-in stats OSD

notgood opened the pull request:
**Building with Lua 5.2 library enables mpv support of builtin stats OSD.**

Lua 5.2 is required to enable mpv player support of builtin stats OSD.
MPV doesn't support later Lua versions, newer versions aren't backwards
compatible, thus including version in a package name.
Upstream Lua doesn't include pkg-config file, including one from ArchLinux.

See https://github.com/mpv-player/mpv/wiki/FAQ#Why_does_mpv_not_support_Lua_53_or_newer

Build and run tested on Allwinner.

heitbaum commented on packages/lang/lua52/package.mk:
```diff
@@ -0,0 +1,27 @@
+  mkdir -p ${SYSROOT_PREFIX}/usr/lib/pkgconfig
+  cp ${PKG_DIR}/config/lua52.pc ${SYSROOT_PREFIX}/usr/lib/pkgconfig
+}
+
```
delete blank line

heitbaum commented on packages/lang/lua52/config/lua52.pc:
```diff
@@ -0,0 +1,20 @@
+V=5.2
```
given that there are versions in this file - would prefer to see the hardcoding version # managed via sed / as part of the makefile ... - otherwise if will be missed.

heitbaum reviewed (commented):
Reviewed the sites at, noting that 5.2.4 is not tagged at the github. 
- But is in the branch: https://github.com/lua/lua/commits/v5-2
- Differences between 5.2.3 and 5.2.4 are https://github.com/lua/lua/compare/v5.2.3...v5-2

https://www.lua.org/ftp/
https://github.com/lua/lua

Checked that there is not a backport option for the PC file.

Is there an alternative .PC solution in another distribution?

notgood commented on packages/lang/lua52/package.mk:
```diff
@@ -0,0 +1,27 @@
+# SPDX-License-Identifier: GPL-2.0
+# Copyright (C) 2019-present Team LibreELEC (https://libreelec.tv)
```
Done

notgood commented on packages/lang/lua52/package.mk:
```diff
@@ -0,0 +1,27 @@
+}
+
+makeinstall_target() {
+  mkdir -p ${SYSROOT_PREFIX}/usr/include/lua5.2
```
Done

notgood commented:
> Is there an alternative .PC solution in another distributio

According to mpv check for lua52, every distro goes it own way with .pc
See https://github.com/mpv-player/mpv/blob/master/waftools/checks/custom.py#L60

Anyway, I've simplified .pc and removed hardcoded versions everywhere.

heitbaum commented:
I did a test build of `mpv-drmprime` - once `waf` was fixed - it builds on both Generic/Allwinner. Did not run test.

notgood commented:
![screenshot](https://user-images.githubusercontent.com/8156459/152226000-8da29c14-96ea-41d4-9c3d-26e525bf74fc.jpg)
Run tested on Allwinner

CvH reviewed (approved):
tx
